mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -841,20 +845,23 @@ impl MakerSwap {
             Ok(res) => match res {
                 Some(tx) => tx,
                 None => {
-                    let payment_fut = self.maker_coin.send_maker_payment(SendPaymentArgs {
-                        time_lock_duration: self.r().data.lock_duration,
-                        time_lock: self.r().data.maker_payment_lock,
-                        other_pubkey: &*self.r().other_maker_coin_htlc_pub,
-                        secret_hash: secret_hash.as_slice(),
-                        amount: self.maker_amount.clone(),
-                        swap_contract_address: &self.r().data.maker_coin_swap_contract_address,
-                        swap_unique_data: &unique_data,
-                        payment_instructions: &self.r().payment_instructions,
-                        watcher_reward,
-                        wait_for_confirmation_until: wait_maker_payment_until,
-                    });
-
-                    match payment_fut.compat().await {
+                    let payment = self
+                        .maker_coin
+                        .send_maker_payment(SendPaymentArgs {
+                            time_lock_duration: lock_duration,
+                            time_lock: maker_payment_lock,
+                            other_pubkey: &*other_maker_coin_htlc_pub,
+                            secret_hash: secret_hash.as_slice(),
+                            amount: self.maker_amount.clone(),
+                            swap_contract_address: &maker_coin_swap_contract_address,
+                            swap_unique_data: &unique_data,
+                            payment_instructions: &payment_instructions,
+                            watcher_reward,
+                            wait_for_confirmation_until: wait_maker_payment_until,
+                        })
+                        .await;
+
+                    match payment {
```

Comment:
I see you’re trying to minimize read access to the RwLock, which is why you’re opting for a deep copy of the data. However, I’d suggest that accessing these fields as they were before without a deep copy could be more efficient option. 
```rust
let maker_coin_swap_contract_address =  self.r().data.maker_coin_swap_contract_address.clone();
let unique_data = self.unique_swap_data();
let payment_instructions = self.r().payment_instructions.clone();
```

wdyt?